oak/server/rust/oak_runtime/src/runtime/tests.rs

```diff
@@ -58,29 +58,111 @@ fn run_node_body(node_label: Label, node_body: Box<NodeBody>) {
     assert_eq!(Ok(()), result);
 }
 
-/// Create a test Node that creates a channel and succeeds.
+/// Returns a non-trivial label for testing.
+fn test_label() -> Label {
+    Label {
+        secrecy_tags: vec![oak_abi::label::authorization_bearer_token_hmac_tag(&[
+            1, 1, 1,
+        ])],
+        integrity_tags: vec![],
+    }
+}
+
+/// Checks that a panic in the node body actually causes the test case to fail, and does not
+/// accidentally get ignored.
 #[test]
-fn create_channel_success() {
+#[ignore]
+#[should_panic]
+fn panic_check() {
+    let label = test_label();
     run_node_body(
-        Label::public_trusted(),
-        Box::new(|runtime| {
+        &label,
+        Box::new(|_runtime| {
+            panic!("testing that panic works");
+        }),
+    );
+}
```

Comment:
@daviddrysdale FYI I am adding this (currently ignored) test in this PR, feel free to un-ignore or remove entirely when you get to the bottom of the synchronization issue.

Reply:
Thanks, I've stolen it for #978